### PR TITLE
RavenDB-9487: Better Low Memory Notifications Handling

### PR DESCRIPTION
**Pull request by aviviadi**
* PrivateWorkingSet is not being calculated correctly - shared mem is calculated (approx)
* Misscalc of freeOrCheapToFree fixed

*Note : Sparrow->System.Diagnostic.Process will use 4.1.0 (and not 4.3.0) to be netstandard1.3 compliant

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -147,7 +147,9 @@ public abstract class Index : IDocumentTombstoneAware, IDisposable, ILowMemoryHa
         private bool _allocationCleanupNeeded;
         private readonly MultipleUseFlag _lowMemoryFlag = new MultipleUseFlag();
         private long _lastLowMemoryEventTicks = 0;
-        private readonly long _lowMemoryIntervalTicks = TimeSpan.FromMinutes(1).Ticks;
+
+        // delay a bit after low mem is over in order to let the flush(es) to finish (or at least do some major flushing writes)
+        private readonly long _lowMemoryIntervalTicks = TimeSpan.FromSeconds(1).Ticks;
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Index.cs:152]: Delete this, it is not required.

File: src/Sparrow/LowMemory/LowMemoryNotification.cs [hunk 2]
```diff
@@ -119,6 +123,46 @@ public LowMemoryNotification(long lowMemoryThreshold)
...
+        {
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                var workingSet =
+                    PlatformDetails.RunningOnPosix == false || PlatformDetails.RunningOnMacOsx
+                        ? currentProcess.WorkingSet64
+                        : MemoryInformation.GetRssMemoryUsage(currentProcess.Id);
+
+                long totalUnmanagedAllocations = 0;
+                foreach (var stats in NativeMemory.ThreadAllocations.Values
```
Comment (ayende) [inline src/Sparrow/LowMemory/LowMemoryNotification.cs:136]: Change this to use a direct call. It is going to be called a lot, let's not do all these allocations

File: src/Sparrow/LowMemory/LowMemoryNotification.cs [hunk 3]
```diff
@@ -119,6 +123,46 @@ public LowMemoryNotification(long lowMemoryThreshold)
...
+                    .Where(x => x.ThreadInstance.IsAlive)
+                    .GroupBy(x => x.Name)
```
Comment (ayende) [inline src/Sparrow/LowMemory/LowMemoryNotification.cs:138]: This is not required., nor is the sorting.

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 1 above)
Comment (aviviadi) [reply src/Raven.Server/Documents/Indexes/Index.cs:152]: It is actually needed.  We saw big perf diff without it

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Index.cs:152]: Okay

**Pull request merged**